specified main()'s return type

bcko opened the pull request:
https://www.dartlang.org/guides/language/language-tour#the-main-function

bcko committed:
specified main()'s return type

https://www.dartlang.org/guides/language/language-tour#the-main-function

devoncarew commented on bin/dart2_fix.dart:
```diff
@@ -7,7 +7,7 @@ import 'dart:io';
 import 'package:dart2_fix/src/model.dart';
 
-main(List<String> args) async {
+void main(List<String> args) async {
```
This should be `Future` :) (because its an async method)

bcko replied:
Oh I see, thank you :)

bcko committed:
changed main() return type to be Future

devoncarew commented:
And, you likely need an 'import 'dart:async`;` now (https://travis-ci.org/dart-lang/dart2_fix/builds/383416131#L572), to reference the `Future`.

bcko committed:
dart:async library is imported

dart:async library import is required in order to specify main()'s return type (Future)

bcko commented:
Thank you for your help! :)

devoncarew commented:
Thanks!

eernstg commented:
Forgive me for commenting on an already closed PR, but here's a bit of information that may be useful in similar situations:

The language team actually decided to allow an async method to have the return type `void`, and the motivation is that this is a meaningful representation of the intent that invocations of that function should be a "fire and forget" operation: Callers should discard the returned future, and this means that they can't even synchronize on the completion of the associated task.

Considering an async `main`, we actually have a good match for this description. The specification says nothing about how the returned value is used, and there is no support for claiming that the returned value is not ignored.

This means that we could make the case that `void main() async {...}` is fine, and in the concrete context of this PR it also allows us to avoid an import.

devoncarew commented:
Awesome, thanks for the context and explanation @eernstg!
